server/repository/repository.go

```diff
@@ -481,14 +482,53 @@ func (s *Server) DeleteRepository(ctx context.Context, q *repositorypkg.RepoQuer
 	}
 
 	// invalidate cache
-	if err := s.cache.SetRepoConnectionState(q.Repo, nil); err == nil {
+	if err := s.cache.SetRepoConnectionState(repo.Repo, repo.Project, nil); err == nil {
 		log.Errorf("error invalidating cache: %v", err)
 	}
 
-	err = s.db.DeleteRepository(ctx, q.Repo)
+	err = s.db.DeleteRepository(ctx, repo.Repo, repo.Project)
 	return &repositorypkg.RepoResponse{}, err
 }
 
+// getRepository fetches a single repository which the user has access to. If only one repository can be found which
+// matches the same URL, that will be returned (this is for backward compatibility reasons). If multiple repositories
+// are matched, a repository is only returned if it matches the app project of the incoming request.
```

Comment:
If only one url matches and it doesn't match the app project of the incoming request, what happens?

the comment implies that the specific appproject is ignored -- the code also does, but this feels surprising.

consider the case where there _were_ two urls that matched 2 minutes ago, I construct a query asking about a specific appproject, and then someone deletes  that specific instance before I send the query.

If I get an answer about something I specifically indicated was outside of my scope, that'd be a pretty surprising response, no?

Reply:
Yeah, the more I think about it we only need to maintain the existing behavior if the app project is empty. Ideally the matching should be strict in all cases, but we cannot do that for backwards compatibility reasons. Since the app project parameter is a new (optional) param for `app repo get` and `app repo delete` we can strictly match the repo if an app project is given.